src/debugpy/_vendored/pydevd/_pydev_bundle/pydev_monkey.py

```diff
@@ -249,99 +230,147 @@ def patch_args(args, is_exec=False):
     try:
         pydev_log.debug("Patching args: %s", args)
         original_args = args
-        args = remove_quotes_from_args(args)
+        unquoted_args = remove_quotes_from_args(args)
+
+        # Internally we should reference original_args (if we want to return them) or unquoted_args
+        # to add to the list which will be then quoted in the end.
+        del args
 
         from pydevd import SetupHolder
-        new_args = []
-        if len(args) == 0:
+        if not unquoted_args:
+            return original_args
+
+        if not is_python(unquoted_args[0]):
+            pydev_log.debug("Process is not python, returning.")
             return original_args
 
-        if is_python(args[0]):
-            ind_c = get_c_option_index(args)
+        # Note: we create a copy as string to help with analyzing the arguments, but
+        # the final list should have items from the unquoted_args as they were initially.
+        args_as_str = _get_str_type_compatible('', unquoted_args)
 
-            if ind_c != -1:
-                host, port = _get_host_port()
+        params_with_flag = ('-W', '-X', '--check-hash-based-pycs', '--jit')
```

Comment:
There's a catch with these:
https://github.com/microsoft/ptvsd/blob/46e1b758bc3694df7b4f08f24b62e73c59494f73/src/ptvsd/multiproc.py#L275-L283

Reply:
I'll take a look at this (quite tricky indeed, even something as `python -qmmodule_to_run` may be done, so, even `-m` and `-c` may be combined).

Reply:
Ok, there's now code to handle the combination of short flags.